File: python/difference_of_squares.py
def square_of_sum(number):
    return sum(i + 1 for i in range(number)) ** 2


def sum_of_squares(number):
    return sum((i+1) ** 2 for i in range(number))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: zig/difference_of_squares.zig
pub fn squareOfSum(n: usize) usize {
    const sum = (1 + n) * n / 2;
    return sum * sum;
}

pub fn sumOfSquares(n: usize) usize {
    return (n * (n + 1) * (2 * n + 1)) / 6;
}

pub fn differenceOfSquares(n: usize) usize {
    return squareOfSum(n) - sumOfSquares(n);
}

